app/listeners/reporting_event_listener.rb

```diff
@@ -90,6 +90,34 @@ def conversation_bot_handoff(event)
     reporting_event.save!
   end
 
+  def conversation_opened(event)
+    conversation = extract_conversation_and_account(event)[0]
+
+    # Find the most recent resolved event for this conversation
+    last_resolved_event = ReportingEvent.where(
+      conversation_id: conversation.id,
+      name: 'conversation_resolved'
+    ).order(event_end_time: :desc).first
+
+    # If there's a previous resolved event, this is a reopening
+    return unless last_resolved_event
+
+    time_since_resolved = conversation.updated_at.to_i - last_resolved_event.event_end_time.to_i
+
+    reporting_event = ReportingEvent.new(
+      name: 'conversation_reopened',
```

Comment:
Should we make it conversation_opened and have it available for every conversation?

Reply:
Makes sense, this can be done